rust/gui-client/src-tauri/src/client/network_changes/windows.rs

```diff
@@ -429,3 +403,337 @@ fn get_apartment_type() -> WinResult<(Com::APTTYPE, Com::APTTYPEQUALIFIER)> {
     unsafe { Com::CoGetApartmentType(&mut apt_type, &mut apt_qualifier) }?;
     Ok((apt_type, apt_qualifier))
 }
+
+mod async_dns {
+    use anyhow::{Context, Result};
+    use std::{ffi::c_void, ops::Deref, path::Path};
+    use tokio::sync::mpsc;
+    use windows::Win32::{
+        Foundation::{CloseHandle, BOOLEAN, HANDLE, INVALID_HANDLE_VALUE},
+        System::Registry,
+        System::Threading::{
+            CreateEventA, RegisterWaitForSingleObject, UnregisterWaitEx, INFINITE,
+            WT_EXECUTEINWAITTHREAD,
+        },
+    };
+    use winreg::RegKey;
+
+    /// Opens and returns the IPv4 and IPv6 registry keys
+    ///
+    /// `HKEY_LOCAL_MACHINE/SYSTEM/CurrentControlSet/Services/Tcpip[6]/Parameters/Interfaces`
+    fn open_network_registry_keys() -> Result<(RegKey, RegKey)> {
+        let hklm = winreg::RegKey::predef(winreg::enums::HKEY_LOCAL_MACHINE);
+        let path_4 = Path::new("SYSTEM")
+            .join("CurrentControlSet")
+            .join("Services")
+            .join("Tcpip")
+            .join("Parameters")
+            .join("Interfaces");
+        let path_6 = Path::new("SYSTEM")
+            .join("CurrentControlSet")
+            .join("Services")
+            .join("Tcpip6")
+            .join("Parameters")
+            .join("Interfaces");
+        let flags = winreg::enums::KEY_NOTIFY;
+        Ok((
+            hklm.open_subkey_with_flags(path_4, flags)?,
+            hklm.open_subkey_with_flags(path_6, flags)?,
+        ))
+    }
+
+    pub(crate) fn run_debug() -> Result<()> {
+        tracing_subscriber::fmt::init();
+        let rt = tokio::runtime::Runtime::new()?;
```

Comment:
Out of curiosity, why make a new runtime here and not just make it an async function?

Reply:
I see! I should have checked the code in more detailed, I was assuming that we use `async fn main`!